bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1355,30 +1355,7 @@ synchronized void updateLastConfirmed(long lac, long len) {
      */
 
     public void asyncReadLastConfirmed(final ReadLastConfirmedCallback cb, final Object ctx) {
-        boolean isClosed;
-        long lastEntryId;
-        synchronized (this) {
-            isClosed = metadata.isClosed();
-            lastEntryId = metadata.getLastEntryId();
-        }
-        if (isClosed) {
-            cb.readLastConfirmedComplete(BKException.Code.OK, lastEntryId, ctx);
-            return;
-        }
-
-        ReadLastConfirmedOp.LastConfirmedDataCallback innercb = new ReadLastConfirmedOp.LastConfirmedDataCallback() {
-                @Override
-                public void readLastConfirmedDataComplete(int rc, DigestManager.RecoveryData data) {
-                    if (rc == BKException.Code.OK) {
-                        updateLastConfirmed(data.getLastAddConfirmed(), data.getLength());
-                        cb.readLastConfirmedComplete(rc, data.getLastAddConfirmed(), ctx);
-                    } else {
-                        cb.readLastConfirmedComplete(rc, INVALID_ENTRY_ID, ctx);
-                    }
-                }
-            };
-
-        new ReadLastConfirmedOp(this, innercb).initiate();
+        asyncReadExplicitLastConfirmed(cb, ctx);
```

Comment:
@sijie here I am changing also *asyncReadLastConfirmed*

This is actually the core of the change.
I had to re implement *asyncTryReadLastConfirmed* because there is no existing support for explicitLAC